public/components/trace_analytics/components/traces/__tests__/span_detail_panel.test.tsx

```diff
@@ -4,55 +4,165 @@
+      hoverinfo: 'none',
+      showlegend: false,
+    },
+  ],
+  table: [
+    {
+      service_name: 'service1',
+      span_id: 'span1',
+      latency: 10,
+      error: 'Error',
+      start_time: '2023-01-01T00:00:00Z',
+      end_time: '2023-01-01T00:00:10Z',
+    },
+  ],
+  ganttMaxX: 20,
+};
+
+const mockSetData = jest.fn();
+const mockAddSpanFilter = jest.fn();
+const mockProps = {
+  http: mockHttp,
+  traceId: 'trace1',
+  colorMap: { service1: '#7492e7' },
+  mode: 'data_prepper',
+  dataSourceMDSId: 'mock-id',
+  dataSourceMDSLabel: 'mock-label',
+  data: mockData,
+  setData: mockSetData,
+  addSpanFilter: mockAddSpanFilter,
+  removeSpanFilter: jest.fn(),
+};
+
+describe('SpanDetailPanel component', () => {
+  it('renders correctly with default props', () => {
+    const wrapper = shallow(<SpanDetailPanel {...mockProps} />);
     expect(wrapper).toMatchSnapshot();
   });
+
+  it('renders gantt chart and mini-map correctly', () => {
+    const wrapper = mount(<SpanDetailPanel {...mockProps} />);
+    expect(wrapper.find(Plt)).toHaveLength(2); // Gantt chart and mini-map
+  });
+
+  it('handles zoom reset button correctly', () => {
+    const wrapper = mount(<SpanDetailPanel {...mockProps} />);
+
+    // Verify that the reset button is initially disabled
+    let resetButton = wrapper
+      .find(EuiSmallButton)
+      .filterWhere((btn) => btn.text().includes('Reset zoom'));
+    expect(resetButton.prop('isDisabled')).toBe(true);
+
+    // Simulate a click on the mini-map
+    const miniMap = wrapper.find('[data-test-subj="mocked-plt"]').at(0);
+    act(() => {
+      miniMap.simulate('click');
+    });
+
+    wrapper.update();
+
+    // Verify that the reset button is now enabled
+    resetButton = wrapper
+      .find(EuiSmallButton)
+      .filterWhere((btn) => btn.text().includes('Reset zoom'));
+    expect(resetButton.prop('isDisabled')).toBe(false); // Should now be enabled
+
+    // Simulate clicking the reset button
+    act(() => {
+      resetButton.prop('onClick')!();
+    });
+
+    wrapper.update();
+
+    // Verify that the reset button is disabled again after reset
+    resetButton = wrapper
+      .find(EuiSmallButton)
+      .filterWhere((btn) => btn.text().includes('Reset zoom'));
+    expect(resetButton.prop('isDisabled')).toBe(true);
+  });
+
+  it('handles view toggle button group', () => {
+    const wrapper = mount(<SpanDetailPanel {...mockProps} />);
+    const toggleButtons = wrapper.find(EuiButtonGroup);
+    expect(toggleButtons).toHaveLength(1);
+
+    // Verify initial state is 'timeline'
+    expect(toggleButtons.prop('idSelected')).toBe('timeline');
+
+    // Simulate changing the toggle
+    act(() => {
+      toggleButtons.prop('onChange')!('span_list');
+    });
+
+    wrapper.update();
+
+    // Verify the toggle button group has been updated
+    const updatedToggleButtons = wrapper.find(EuiButtonGroup);
+    expect(updatedToggleButtons.prop('idSelected')).toBe('span_list');
+  });
```

Comment:
For this test I feel we are testing the button and not what our page is rendering on button click. The button is already tested in OUI, we should rather test on button change is the rendered component changing to gantt-chart, table or tree view.

Reply:
Removed the test as this functionality is covered by OUI and the table specific checks should be done in span_detail_table